src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,214 @@
+import { createUniqueIDFactory } from '@shopify/javascript-utilities/other';
+import { findFirstFocusableNode } from '@shopify/javascript-utilities/focus';
+import { addEventListener, removeEventListener } from '@shopify/javascript-utilities/events';
+
+import DropdownItem from './DropdownItem';
+import { classNames } from '@shopify/react-utilities/styles';
+import { DROPDOWN } from '../ThemeIdentifiers';
+import * as baseTheme from './Dropdown.scss';
+import { Keys } from '../../types';
+import Button from '../Button';
+
+export type Direction = 'up' | 'down' | 'left' | 'right';
+
+export interface DropdownItemProps {
+  children?: React.ReactNode;
+  disabled?: boolean;
+  divider?: boolean;
+  header?: boolean;
+  onClick?(): void;
+}
+
+export interface Props {
+  disabled?: boolean;
+  children?: string;
+  direction?: Direction;
+  active: boolean;
+  activatorWrapper?: string;
+  DropdownItems: DropdownItemProps[];
+  onClose?(): void;
+  toggle?() : void;
+}
+
+export interface State {
+  selectedIndex: number
+}
+
+const getUniqueID = createUniqueIDFactory('Dropdown');
+
+export class Dropdown extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { 
+      selectedIndex: 0 
+    };
+  }
+
+  private activatorContainer: HTMLElement | null;
+
+  private id = getUniqueID();
+
+  componentWillReceiveProps(newProps: any) {
+    const { active } = this.props;
+
+    if (active && !newProps.active && typeof newProps.onClose !== 'undefined') {
+      newProps.onClose();
+    }
+  }
+
+  componentDidMount() {
+    this.setAccessibilityAttributes();
+    addEventListener(document, 'keyup', this.handleKeyEvent);
+  }
+
+  componentDidUpdate() {
+    this.setAccessibilityAttributes();
+  }
+
+  componentWillUnmount() {
+    removeEventListener(document, 'keyup', this.handleKeyEvent);
+  }
+
+  render() {
+    const { 
+      activatorWrapper: WRAPPERCOMPONENT = 'div',
+      active,
+      children,
+      DropdownItems,
+      toggle,
+      direction,
+      disabled
+    } = this.props;
+    
+    const {
+      selectedIndex
+    } = this.state;
+
+    
+    const dropdownClassName = classNames (
+      typeof direction == 'undefined' || direction === "down" ? baseTheme.dropdown 
+        : direction === "up" ? baseTheme.dropup
+        : direction === "left" ? baseTheme.dropleft
+        : baseTheme.dropright,
+       !disabled && active && baseTheme.active
+    );
+
+    const dropdownMenuClassName = classNames (
+      baseTheme.dropdownMenu,
+      !disabled && active && baseTheme.active,
+    );
+
+    const DropdownItemComponents = DropdownItems.map((item,index) => 
+            <DropdownItem 
+              key={index}
+              active={selectedIndex === index} 
+              disabled={item.disabled}
+              divider={item.divider}
+              header={item.header}
+              children={item.children}
+            ></DropdownItem>
+        );
```

Comment:
@shyamk9 : Make this as self ending tag, as it doesn't have anything inside the <DropdownItem>